lib/solar_system.rb

```diff
@@ -0,0 +1,58 @@
+class SolarSystem
+
+  attr_reader :star_name, :planets
+
+  def initialize(star_name)
+
+    unless star_name.instance_of?(String) && !star_name.empty?
+      raise ArgumentError, 'Star Name must be valid string'
+    end
+
+      @star_name = star_name
+    @planets = []
+  end
+
+  def add_planet(planet)
+
+    unless planet.instance_of?(Planet)
+      raise ArgumentError, "Planet must be instance of Planet class"
+    end
+
+    @planets.each do |established_planet|
+      if established_planet.name == planet.name
+        raise ArgumentError, "Planet #{planet} is already in planet list"
+      end
+    end
+
+    @planets << planet
+  end
+
+  def list_planets
+    str = "Planets orbiting #{@star_name}"
+
+    @planets.each_with_index do |planet, i|
+      str += "\n #{i+1}: #{planet.name}"
+    end
+
+    return str
+  end
+
+  def find_planet_by_name(planet_name)
+    planet = @planets.find {|planet| planet.name == planet_name}
+
+    unless planet
+      raise ArgumentError, "Planet #{planet_name} is not found in list"
+    end
```

Comment:
I like that you've been working with exceptions throughout this project - this is good practice.

In this particular case, I'm don't think an exception is the right choice. In practice, a `find`-type method typically returns `nil` if whatever you were looking for wasn't found. See [the documentation for `Array#find`](https://ruby-doc.org/core/Enumerable.html#method-i-find) for an example.